### PR TITLE
Update migration guide, add framework feature comparison

### PR DESCRIPTION
**Pull request by provinzkraut**
# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [ ] Have you got 100% test coverage on new code?
- [x] Have you updated the prose documentation?
- [ ] Have you updated the reference documentation?

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -24,7 +24,8 @@
 </center>
 <!-- markdownlint-restore -->
 
-Starlite is a light, opinionated and flexible ASGI API framework built on top of **[pydantic](https://github.com/samuelcolvin/pydantic)**.
+Starlite is a light, opinionated and flexible ASGI API framework built on top
```
Comment (Goldziher) [inline docs/index.md:27]: I'm not sure we're builtin on top of. I think we can drop this and simply add we support pydantic 100%. The internals are internal.

File: docs/index.md [hunk 2]
```diff
@@ -103,7 +104,8 @@ class User(BaseModel):
     id: UUID4
 ```
 
-Alternatively, you can **use a dataclass** – either from dataclasses or from pydantic, or a [`TypedDict`][typing.TypedDict]:
+Alternatively, you can **use a dataclass** – either from dataclasses or from pydantic, or
+a [`TypedDict`][typing.TypedDict]:
```
Comment (Goldziher) [inline docs/index.md:108]: Also msgspec.Struct

File: docs/index.md [hunk 3]
```diff
@@ -182,26 +184,50 @@ uvicorn my_app.main:app --reload
...
+The idea to build an ASGI framework that's deeply integrated with pydantic is of course not new - it was first done in
+FastAPI,
+which in this regard (and some others) was a source of inspiration for this framework. Nonetheless, Starlite is not
+FastAPI -
 it has a different design, different project goals and a completely different codebase.
 
 1. The goal of this project is to be community-driven. That is, not to have a single author,
    but rather a core team of maintainers leading the project, as well as community contributors.
    Starlite currently has 5 maintainers and is being very actively developed.
 2. Starlite draws **inspiration from NestJS** - a contemporary TypeScript framework - which places opinions and patterns
```
Comment (Goldziher) [inline docs/index.md:200]: I think we can drop this part.

File: docs/migration.md [hunk 4]
```diff
@@ -1,58 +1,361 @@
 # Migration to Starlite
 
-Migrating **from either Starlette or FastAPI** to Starlite is rather uncomplicated, because the frameworks are for the most
+Migrating **from either Starlette or FastAPI** to Starlite is rather uncomplicated, because the frameworks are for the
```
Comment (Goldziher) [inline docs/migration.md:3]: Simple instead of rather uncomplicated?

**Review by Goldziher — APPROVED** [review]
👍👍👍 😃

File: docs/migration.md (hunk 4 above)
Comment (provinzkraut) [reply docs/migration.md:3]: That would be too uncomplicated :P
I'll reword the prose sections when I'm done with the rest.

**Pull request merged**